src/modules/site-settings/edit-site-details.tsx

```diff
@@ -45,6 +46,11 @@ export default function EditSiteDetails( { currentWpVersion, onSave }: EditSiteD
 		( selectedSite?.phpVersion as AllowedPHPVersion ) ?? DEFAULT_PHP_VERSION
 	);
 	const [ selectedWpVersion, setSelectedWpVersion ] = useState( currentWpVersion );
+	const [ useCustomDomain, setUseCustomDomain ] = useState( Boolean( selectedSite?.customDomain ) );
```

Comment:
It seems a lot of this logic is very similar to the one we added to `site-form`
I believe there's an opportunity here to create a shared component/hook, to avoid repetition.